### PR TITLE
Upgrade to PHPUnit 9

### PR DESCRIPTION
**Pull request by dennisameling**
Work in progress ⚠️ 

**Prerequisites for this PR:**
- ✔️ We updated PHPSTAN to `0.12.25` in https://github.com/mautic/mautic/pull/9522 which is needed for some dependency updates 
- ~~⏳ Currently waiting for https://github.com/mautic/mautic/pull/9451 to be merged - I can already work on this PR, but #9451 will need to be merged first before we merge this one 👍🏼~~

**Some misc issues I fixed in this PR:**
- ⚠️ Don't use Mautic's ErrorHandler when running in PHPUnit https://github.com/mautic/mautic/pull/9602/commits/56f82e46ac75086ad79af69d67c4e5f77caa9b58 - this allows PHPUnit to convert PHP notices/warnings/etc. into exceptions. We've been having a bunch of notices/warnings/etc. that PR authors aren't notified about because PHPUnit can't convert them into exceptions - example: https://github.com/mautic/mautic/runs/1735649866
- Bunch of PHP notices/warnings in PHPUnit tests that have been there for a while:
  - `PHP Notice:  Undefined index: show_php_version_warning_if_under`
  - `PHP Warning:  exif_imagetype(path/to/file/1/fieldId1/upload1): failed to open stream: No such file or directory`
  - `Constant MAUTIC_TABLE_PREFIX already defined`
  - `PHP Notice:  Undefined index:  in /home/runner/work/mautic/mautic/app/bundles/ApiBundle/Helper/BatchIdToEntityHelper.php`
  - `PHP Notice:  Undefined offset: 1 in /home/runner/work/mautic/mautic/app/bundles/SmsBundle/EventListener/SmsSubscriber.php on line 134`
- Codecov warning: `Issue detecting commit SHA. Please run actions/checkout with fetch-depth > 1 or set to 0` https://github.com/mautic/mautic/pull/9602/commits/268549e6f4da5814922d2519c7b52fb1d85273ab

### CODE REVIEW

**Comment by dennisameling** [comment]
@escopecz I have PHPUnit 9 working now (see https://github.com/mautic/mautic/pull/9602/checks?check_run_id=1741236093), but there's a bunch of deprecation warnings for PHPUnit 10. Mostly `The at() matcher has been deprecated`. Is this something we want to fix in this PR as well or could we leave it to a later point in time? What do you think? [We can't disable](https://github.com/sebastianbergmann/phpunit/issues/4228#issuecomment-629030831) the deprecation warnings unfortunately.

**Comment by escopecz** [comment]
Deprecations can be handled in a separate PR.

File: app/AppKernel.php
Comment (dennisameling) [inline app/AppKernel.php:209]: Newer versions of the PsyshBundle have Symfony 4+ as a hard dependency. Will fix this in the Symfony 4 PR as soon as this one is merged https://github.com/mautic/mautic/pull/9409

File: app/bundles/EmailBundle/Tests/Controller/Api/EmailApiControllerFunctionalTest.php
Comment (dennisameling) [inline app/bundles/EmailBundle/Tests/Controller/Api/EmailApiControllerFunctionalTest.php:138]: DELETE requests don't look at body parameters in Mautic's code, only at query parameters. We recently added this to the docs, at least for batch deleting contacts https://github.com/mautic/developer-documentation/pull/177/files

File: app/phpunit.xml.dist
Comment (dennisameling) [inline app/phpunit.xml.dist:2]: PHPUnit indicated that its XML config was deprecated, so I ran their built-in command `bin/phpunit --migrate-configuration` to update it

File: composer.json
Comment (dennisameling) [inline composer.json:134]: AFAIK we're not using Selenium but are looking into Cypress instead now, so I removed the dev dependency.

**Review by escopecz — APPROVED** [review]
:+1: Thanks Dennis!

**Pull request merged**
